### PR TITLE
Feature/202/First login popup

### PR DESCRIPTION
**Pull request by High-Voltaged**
- An `isFirstLogin` field was added to the redux state;
- A check for that field's value was added before rendering a popup on the mentor page;
- Tests for the `MentorHome` component were added.

### CODE REVIEW

File: tests/unit/pages/mentor-home/MentorHome.spec.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { ModalProvider } from '~/context/modal-context'
+import { ConfirmationDialogProvider } from '~/context/confirm-context'
+import MentorHome from '~/pages/mentor-home/MentorHome'
+
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn()
+}))
+
+const MentorHomeWithProviders = () => (
+  <MemoryRouter>
```
Comment (tarasSanotskyi) [inline tests/unit/pages/mentor-home/MentorHome.spec.js:14]: You don`t need memory router. you can use renderWithProviders util it include Memory Router and Provider

File: tests/unit/pages/mentor-home/MentorHome.spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      <ModalProvider>
+        <MentorHome />
+      </ModalProvider>
+    </ConfirmationDialogProvider>
+  </MemoryRouter>
+)
+
+describe('MentorHome component', () => {
+  it('should render a BecomeATutor modal when logging in for the first time', () => {
+    useSelector.mockImplementation((fn) =>
```
Comment (tarasSanotskyi) [inline tests/unit/pages/mentor-home/MentorHome.spec.js:25]: test redux with Provider, and remove this mocks for useSelector

**Review by tarasSanotskyi — APPROVED** [review]

File: tests/unit/pages/mentor-home/MentorHome.spec.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  it("shouldn't render a BecomeATutor modal when logging in for the second time", () => {
+    useSelector.mockImplementation((fn) =>
+      fn({
+        appMain: { isFirstLogin: false }
+      })
+    )
+    render(<MentorHomeWithProviders />)
+
+    const titleToFind = /becomeTutor.generalInfo.title/i
+    expect(() => screen.getByText(titleToFind)).toThrow()
```
Comment (rostyslaw27) [inline tests/unit/pages/mentor-home/MentorHome.spec.js:45]: const titleToFind = screen.queryByText(/becomeTutor.generalInfo.title/i)
expect(titleToFind).not.toBeInTheDocument()

**Review by rostyslaw27 — CHANGES_REQUESTED** [review]

File: tests/unit/pages/mentor-home/MentorHome.spec.js (hunk 2 above)
Comment (rostyslaw27) [reply tests/unit/pages/mentor-home/MentorHome.spec.js:25]: we need to test with different isFirstLogin values

File: tests/unit/pages/mentor-home/MentorHome.spec.js (hunk 3 above)
Comment (High-Voltaged) [reply tests/unit/pages/mentor-home/MentorHome.spec.js]: fixed

File: tests/unit/pages/mentor-home/MentorHome.spec.js (hunk 2 above)
Comment (tarasSanotskyi) [reply tests/unit/pages/mentor-home/MentorHome.spec.js:25]: ok

File: tests/unit/pages/mentor-home/MentorHome.spec.js (hunk 1 above)
Comment (High-Voltaged) [reply tests/unit/pages/mentor-home/MentorHome.spec.js]: fixed

**Review by tarasSanotskyi — APPROVED** [review]

**Review by rostyslaw27 — APPROVED** [review]

File: src/pages/mentor-home/MentorHome.js [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
 const MentorHome = () => {
   const { setModal, setFullScreen } = useContext(ModalContext)
+  const { isFirstLogin } = useSelector((state) => state.appMain)
 
   useEffect(() => {
-    setFullScreen(true)
-    setModal(<BecomeATutor />)
-  }, [setModal, setFullScreen])
+    if (isFirstLogin) {
+      setFullScreen(true)
```
Comment (maarekua) [inline src/pages/mentor-home/MentorHome.js:12]: How fullscreen looks like ?

Comment (maarekua) [reply src/pages/mentor-home/MentorHome.js:12]: I don't think we need a "useEffect" hook here.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/pages/mentor-home/MentorHome.js (hunk 4 above)
Comment (High-Voltaged) [reply src/pages/mentor-home/MentorHome.js:12]: The hook was used before in code, to show the modal on render of this component. Now I need to check for this condition on render of this component. How can I do that without useEffect? I just didn’t know there were other ways

Comment (maarekua) [reply src/pages/mentor-home/MentorHome.js:12]: Place this if in the component without hooks at all :)

Comment (High-Voltaged) [reply src/pages/mentor-home/MentorHome.js:12]: I tried adding it, but there was a “Maximum depth exceeded error”, which I assume was due to multiple rerendering, so I guess I can’t really avoid using useEffect here

Comment (maarekua) [reply src/pages/mentor-home/MentorHome.js:12]: Let's discuss it on our sync-up.

**Review by LubomyrKashuba — APPROVED** [review]

**Review by maarekua — APPROVED** [review]

**Pull request merged**